thorlcr/thorutil/thbuf.cpp

```diff
@@ -2151,21 +2151,9 @@ class CSharedWriteAheadDisk : public CSharedWriteAheadBase
     }
     virtual unsigned __int64 getStatistic(StatisticKind kind) const override
     {
-        switch (kind)
-        {
-            case StSizeSpillFile:
-                return tempFileIO->getStatistic(StSizeDiskWrite);
-            case StCycleDiskWriteIOCycles:
-            case StTimeDiskWriteIO:
-            case StSizeDiskWrite:
-                return 0;
-            case StNumSpills:
-                return 1;
-            case StTimeSpillElapsed:
-                return tempFileIO->getStatistic(StCycleDiskWriteIOCycles);
-            default:
-                return tempFileIO->getStatistic(kind);
-        }
+        if (kind==StNumSpills)
+            return 1;
```

Comment:
is the calling mergeStats called from gatherActiveStats periodically, going to cause StNumSpills to increase every time?

Reply:
No, every time gatherActiveStats is called, it starts with a fresh CRuntimeStatisticCollection (see CSlaveActivity::serializeStats).

Reply:
right, yes